crates/miden-lib/asm/miden/note.masm

```diff
@@ -206,11 +207,41 @@ export.compute_inputs_commitment
     # => [inputs_ptr, num_inputs, pad_inputs_flag]
 
     exec.rpo::prepare_hasher_state
-
     exec.rpo::hash_memory_with_state
     # => [COMMITMENT]
 end
 
+#! Builds the recipient hash from note inputs, script root, and serial number.
+#!
+#! This procedure computes the commitment to the note inputs and then builds the recipient
+#! hash using the provided script root and serial number.
+#!
+#! Inputs:  [inputs_ptr, num_inputs, SERIAL_NUM, SCRIPT_ROOT]
+#! Outputs: [RECIPIENT]
+#!
+#! Where:
+#! - inputs_ptr is the memory address where the note inputs are stored.
+#! - num_inputs is the number of input values.
+#! - SCRIPT_ROOT is the script root of the note.
+#! - SERIAL_NUM is the serial number of the note.
+#! - RECIPIENT is the computed recipient hash.
+#!
+#! Panics if:
+#! - inputs_ptr is not word-aligned (i.e., is not a multiple of 4).
+#! - num_inputs is greater than 128.
+#!
+#! Invocation: exec
+export.build_recipient
+    exec.compute_inputs_commitment
+    # => [INPUTS_HASH, SERIAL_NUM, SCRIPT_ROOT]
+
+    movdnw.2
+    # => [SERIAL_NUM, SCRIPT_ROOT, INPUTS_HASH]
+
+    exec.tx::build_recipient_hash
+    # => [RECIPIENT]
+end
```

Comment:
It feels a bit weird that `build_recipient` is in the `miden::note` module but `build_recipient_hash` is in the `miden::tx` module. These should probably be in the same module, and we have a few options here;

1. Put both into the `miden::tx` module - since these are related to notes, I don't think like this is a good idea.
2. Put both into the `miden::note` module - this is better, but `miden::note` is meant to be for procedures about currently executing notes.
3. Put both into `miden::output_note` - this better than option 1 but not sure if it is better than option 2.
4. Put both into a new module - maybe `miden::note_utils`.

My preference is either 2 or 4.

Reply:
I am fine with this approach (and in this case `build_recipient` stays in `miden::note`), but I also don't mind having the `utils` method. So, both approaches work.